modules/core/src/main/scala/org/scalasteward/core/forge/ForgeType.scala

```diff
@@ -38,35 +42,64 @@ sealed trait ForgeType extends Product with Serializable {
 }
 
 object ForgeType {
+  trait DiffUriPattern { def forDiff(from: String, to: String): Uri => Uri }
+  trait FileUriPattern { def forFile(fileName: String): Uri => Uri }
```

Comment:
`FileUriPattern` & `DiffUriPattern` have been designed so that they can be concisely implemented using the [lambda syntax for Single Abstract Method (SAM) types](https://www.scala-lang.org/news/2.12.0/#lambda-syntax-for-sam-types), eg:

https://github.com/scala-steward-org/scala-steward/blob/13380ebdca515c0bebbb2afc732a76b4b7e30904/modules/core/src/main/scala/org/scalasteward/core/forge/ForgeType.scala#L88-L89

Reply:
![](https://media2.giphy.com/media/QTAVEex4ANH1pcdg16/giphy.gif?cid=ecf05e47igcxisi0f6nrly03nht084c3kdgjxm8bhmvr8cbf&ep=v1_gifs_search&rid=giphy.gif&ct=g)